### PR TITLE
Fixes #12554 - Use proper Operating System name representation

### PR DESCRIPTION
**Pull request by oshtaier**
We should always have entity name to be used for column 'Name', not the description (even if it exists). Seems more logical to fix in that helper method rather than in '..index' page.

### CODE REVIEW

**Comment by tbrisker** [comment]
[test] - This will trigger the automatic testing for this PR.

**Comment by domcleal** [comment]
This seems to revert c2c3240, which was a deliberate attempt to improve the UI.  Why should this be changed back?

**Comment by oshtaier** [comment]
Just few thoughts:
1) If we have column 'Name' - so we should always see entity name here, no matter what intention we can have. Just imagine that we put into search field 'name=AAAAA', but get entity with name 'BBBBB', only because that its description. Seems as really confusing thing and is a defect definitely
2) Even comment to `def to_label` is saying - `# The OS is usually represented as the concatenation of the OS and the revision`, not the _description_ or something.
3) If things were really going to be improved and we thought to use description in some way, so we should add separate column with description explicitly.

So basically, from QA or UI/UX point of view - name is name and description is a description. It is bad practice to mix these completely different entities together

Please, let me know if I need to provide more details

**Comment by domcleal** [comment]
That doesn't seem reason enough to change it and lose this feature (partially too, since the PR makes the UI more inconsistent elsewhere).  If you're going to propose a change, I think the original change needs to be retained.

**Comment by oshtaier** [comment]
Not sure that I understand what you mean. What is a feature that we are going to lose? This PR will not revert anything. We still have `description` field that can be used for 'create' or 'update' procedures. Only one thing that we are changing is a representation of entity name in `Name` column. It is a fix for obvious defect, not removing of functionality and opposite to your words - making UI consistent.

**Comment by domcleal** [comment]
The description column usually contains the LSB name through fact imports, which is the more user friendly name.  Changing it to display the name instead of description hides the LSB name and shows the arbitrary internal/facter name instead, so less user-friendly.

This also makes the OS list inconsistent with the host list and form pages, and probably others (such as multiselect associations) that show the label by default.

**Comment by oshtaier** [comment]
> The description column usually contains the LSB name through fact imports, which is the more user friendly name. Changing it to display the name instead of description

We do not have description column for operating system entity. I am going to display name in `Name` column, not the description. I can't find other words to explain that fact that if user has OS entity with name = 'AAAA', so he/she expects to see it in 'Name' column. Only 'AAAA', nothing else, otherwise it is a mess. I can't imagine situation when that approach will be user friendly.

Also, I can't find other entity in the application that show something in 'Name' column instead name of entity

I am not insisting that my PR is a good and probably I/we need find another way to fix that issue, but I need to understand. Do you not agree with that PR or with a fact that name should be displayed in `Name` column and nothing else?

**Comment by domcleal** [comment]
> We do not have description column for operating system entity

Database column, not UI column, i.e. model attribute.  See earlier link to the commit that adds it.

You're getting hung up on the name of the column and insisting that because it's titled "Name" that it must contain the value of the "Name" attribute.  Call it something else then?

**Comment by oshtaier** [comment]
As an user of that application - I have no idea about database and basically, I shouldn't have. Renaming column in application is not good idea as that will introduce even more inconsistency - we have 'Name' column for every entity. I am not sure how user should understand the fact that one column can represent different filed data depending on the existence of that data in the field (like if you have description - show the description, if you don't - then name, and tomorrow we will show label, in case we don't have description or name). Once more, from UX point - it is a defect as user will never expect behavior like this.
I am not talking about technical aspects of implementation and basically I don' care about them as I am looking on the problem from another perspective. I just gave an idea that works for me and seems logical and not only for me.
And yes - if we provide name for 'Name' column (which is mandatory as for me for every entity) in 100% of cases (please, correct me if I wrong) then it is expected behavior for that situation too.

I completely understand your concerns, but please understand mine either.

**Comment by domcleal** [comment]
> I completely understand your concerns, but please understand mine either.

I do, but this PR doesn't address my review comments as the proposed change is contrary to the functionality added via #3720.  It shouldn't just be removed without equivalent replacement.

> Once more, from UX point - it is a defect as user will never expect behavior like this.

Indeed, it's odd behaviour on this one page, but what's displayed in the column is currently consistent with OS labels in the rest of the application.

The only exception that I can see is that the column heading is inconsistent with the labels on the OS form.  I'd also note that the description should also always be similar to the name in any case, as it's typically auto-generated.

**Comment by oshtaier** [comment]
I see. Now it's make more sense. So it seems that we have at least two issues there. I will try to look on that in case it will not be too complex for me :)

**Comment by dLobatog** [comment]
Just changing the title of the column would do if that's the defect you're finding here? I agree changing it application-wide will fix this at the cost of breaking it everywhere else.

**Comment by oshtaier** [comment]
To be honest, I do not see what else can be changed without touching generic architecture

**Comment by dLobatog** [comment]
[test]

**Comment by domcleal** [comment]
Thanks, could you just update the commit message to reflect what the commit's doing?  It doesn't describe it right now.  If there's not enough room, please keep the first line short and put more info in the description (http://theforeman.org/handbook.html#Codingstandards).

**Comment by oshtaier** [comment]
@domcleal Thanks for pointing out on that. Fixed

**Comment by domcleal** [comment]
[test]

**Comment by domcleal** [comment]
Merged as 6308a7b, thanks @oshtaier.
